receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet.go

```diff
@@ -13,82 +13,182 @@ import (
 
 	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
 	cExtractor "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/k8swindows/extractors"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/kubeletutil"
+
 	"go.uber.org/zap"
 	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
 )
 
-type kubeletSummaryProvider struct {
-	logger   *zap.Logger
-	hostIP   string
-	hostPort string
-	client   *kubeletutil.KubeletClient
-	hostInfo host.Info
+// KubeletProvider Represents interface to kubelet.
+type KubeletProvider interface {
```

Comment:
Are we just changing this to make it unit testable?

Reply:
This PR has 2 changes.
1. Changes to include metric collection at container level.
2. unit tests for kubelet.go
3. Refactor in kubelet to enable unit tests. This interface is created just for purpose of enabling unit tests. I want to separate out metric collection and source of these metrics so that i can mock source of metrics to unit test metric collection code.